packages/deployment/task-reaper.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+set -e
+
+echo "$(date) Starting task-reaper.sh"
+
+INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+
+CLUSTER=$(grep 'ECS_CLUSTER=' /etc/ecs/ecs.config | cut -d '=' -f 2)
+
+CONTAINER_INSTANCE_ARN=$(aws ecs list-container-instances --cluster "$CLUSTER" --filter "ec2InstanceId == $INSTANCE_ID" | jq -r '.containerInstanceArns[0]')
+CONTAINER_INSTANCE_STATUS=$(aws ecs describe-container-instances --cluster "$CLUSTER" --container-instances "$CONTAINER_INSTANCE_ARN" | jq -r .containerInstances[0].status)
+
+if [ "$CONTAINER_INSTANCE_STATUS" = 'ACTIVE' ]; then
+  LIFECYCLE_STATE=$(aws autoscaling describe-auto-scaling-instances --instance-ids "$INSTANCE_ID" | jq -r '.AutoScalingInstances[0].LifecycleState')
+
+  if [ "$LIFECYCLE_STATE" = 'Terminating:Wait' ]; then
+    echo "Draining ${INSTANCE_ID}"
+    aws ecs update-container-instances-state --cluster "$CLUSTER" --container-instances "$CONTAINER_INSTANCE_ARN" --status DRAINING
+  else
+    echo "Nothing to be done, exiting"
+  fi
+else
+  TASKS_COUNT=$(aws ecs list-tasks --cluster "$CLUSTER" --container-instance "$CONTAINER_INSTANCE_ARN" | jq -r '.taskArns | length')
+
+  if [ "$TASKS_COUNT" -eq "0" ]; then
+    ASG_NAME=$(aws autoscaling describe-auto-scaling-instances --instance-ids "$INSTANCE_ID" | jq -r '.AutoScalingInstances[0].AutoScalingGroupName')
+    LIFECYCLE_HOOK_NAME=$(aws autoscaling describe-lifecycle-hooks --auto-scaling-group-name "$ASG_NAME" | jq -r '.LifecycleHooks[0].LifecycleHookName')
```

Comment:
Would this break if we have more than one `LifecycleHooks`?

Reply:
I did some testing locally.
1. Add some lifecycle hook
2. Add another lifecycle hook
3. Run `describe` portion of the above command.

It looks like this _could_ cause issues if there are more than one `LifecycleHook`s attached to the asg, especially if the first one returned is not `EC2_INSTANCE_TERMINATING`.

If having more than one isn't a use-case, or is out of scope, then I'd at least like to have a ticket created so we can circle back.

Reply:
Fixed in 26cbc760f